integration/plugin/volumes/mounts_test.go

```diff
@@ -0,0 +1,58 @@
+package volumes
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration-cli/daemon"
+	"github.com/docker/docker/integration-cli/fixtures/plugin"
+	"github.com/gotestyourself/gotestyourself/assert"
+	"github.com/gotestyourself/gotestyourself/skip"
+)
+
+// TestPluginWithDevMounts tests very specific regression caused by mounts ordering
+// (sorted in the daemon). See #36698
+func TestPluginWithDevMounts(t *testing.T) {
+	skip.IfCondition(t, testEnv.IsRemoteDaemon(), "cannot run daemon when remote daemon")
+	t.Parallel()
+
+	d := daemon.New(t, "", dockerdBinary, daemon.Config{})
+	d.Start(t, "--iptables=false")
+	defer d.Stop(t)
+
+	client, err := d.NewClient()
+	assert.Check(t, err)
```

Comment:
`assert.NilError`

Reply:
This is a copy-paste from the logging package... I don't know why it was changed to `assert.Check`... @dnephin ?

Reply:
I guess `assert.Check` just returns ok if the passed in value is nil. If it's not nil there's some switch on the type, and error types would fail the test.

Reply:
Yes, that's right. The automated migration converted `testify/assert.NoError` (which uses `t.Fail()`) to `assert.Check()` (which uses `t.Fail()`) because those have the equivalent behaviour. Checking errors with `t.Fail()` is generally going to be wrong and lead to a panic later in the test when something else is nil. So the original test was wrong to use `assert.NoError`, and should have used `require.NoError`.

`gotestyourself/assert.NilError(t, err)` is correct because it uses `t.FailNow()` and stops execution of the test. 

I didn't fix ever place that was using the wrong assertion, because there were many, but I tried to fix up a few.